### DISCUSSION TITLE
Localization Source Generator

### DISCUSSION THREAD

**Discussion by ds5678** [ByronMayne/SourceGenerator.Foundations#34] [Show and tell]
Thank you for this wonderful library. It alleviated the greatest pain point I've experienced so far with source generators.

I used your library to help me write a source generator that generates a localized text API from a set of json files. Previously, I had been embedding them in the assembly and deserializing at runtime. Now, my codebase is cleaner, faster, and more responsive to changes.

https://github.com/AssetRipper/AssetRipper/commit/6d01f742623051998e8fb3aefa53342f0e460da2

**Comment by ByronMayne** [comment]
Hey there 👋,

SourceGenerator.Foundations was a tough one to write. It does quite a bit under the scenes with MSBuild juggling, runtime assembly resolving and resource extraction, inter op with DevEnv which ended up requiring a ton of iteration. Hopefully all this complexity is hidden from the end users and it just 'works'.   I wrote it originally for me since I love source generators, but the fact that the error handling was so trash and referencing any NuGet packages was just incredibly painful, I needed to fix that. 

I am glad to see it get adopted by others and help full the massive hole in the development workflows.

Thanks for sharing! 


Also if you have anything that tripped you up while using it, please share :)

**Reply by ds5678** [reply]
> Also if you have anything that tripped you up while using it, please share :)

One of the things I noticed is that I can't use `EnforceAnalyzerRules` with your library because of references to `System.Console`. Other than that, things seem to be going well. I'll almost certainly implement more generators.

**Reply by ByronMayne** [reply]
Let me fix that one, it's pretty easy

**Reply by ByronMayne** [reply]
The fix is in the latest release, the only change was the fix for this. https://github.com/ByronMayne/SourceGenerator.Foundations/releases/tag/2.0.9

It was super painful to get the unit test working on this but I managed to git it, it won't happen again in the future. Thanks for letting me know :)

**Reply by ds5678** [reply]
Thank you for implementing it! I bumped to the latest version, and everything seems great.

**Comment by ByronMayne** [comment]
Maybe I should add a section to the README of projects that use it that are public.

**Reply by ds5678** [reply]
It doesn't matter either way to me; I just wanted to convey my appreciation.
